### ISSUE TITLE
clearOptions should not clear items

### ISSUE THREAD

**Issue by ghost** [selectize/selectize.js#593]
The clearOptions() method results in the clear() method also being called. Given the logical distinction between "options" and "items" this seems incorrect. The behavior also prevents (unless I'm missing another approach) the use case of providing a new set of options for each item, as follows:
    $('#select-state').selectize({
        load: function(query, callback){ 
            this.clearOptions();
            //fetch the new options
        }
    });

**Comment by dresende** [comment]
This is nonsense why it's still open. You should just remove `.clear()` at the end of the function and it just works as expected.

**Comment by dipbazz** [comment]
Has this issue been fixed? if not then how can we implement clearOptions with a hacky way so that our options will be cleared without being cleared the selected input option? If Somebody can help then it would be really appreciated.

**Comment by jicihome** [comment]
According to this changelog:
Fixed bug making clearOptions function. Now it doesn't remove already selected options.

(thanks @caseymct - #1079)

But it doesn't seem to work.
Also, silent is implemented and doesn't see to work too.

**Comment by jicihome** [comment]
I can confirm for silent. Not working for add item

**Comment by KES777** [comment]
This is wrong proposition. `clearOptions` must call `clear` otherwise data will be inconsistent. 

For example you have list 1,2, user selected 7. Is this expected? No, because there is no 7 on the list.  
Same with clearOptions. If options are cleared, you can choose nothing, thus `selected list` also must be cleared.

**Comment by suchoss** [comment]
Imho, this is bad design - method is called `clearOptions` and that is exactly what is expected from the method to do. 
It shouldn't delete any already selected items. 

For example in my case - I need to delete all options all the time, because I am calling backend with every pressed letter on keyboard. Then user selects what they need and I close options and delete them - ready for next query. When they select everything they need, they submit results with button. In version `0.12.6` this worked quite well.

**Comment by jthakilla9** [comment]
> This is nonsense why it's still open. You should just remove `.clear()` at the end of the function and it just works as expected.

Its true what he said, remove the clear and use it like this example:  

it works like butter 👍 

// Functie om gebruikers op te halen en dropdown te vullen met behulp van Selectize
function loadUsers(selectedEmails = []) {
  Fliplet.DataSources.connect(usersDataSourceId)
    .then(connection => connection.find())
    .then(users => {
      if (!users.length) {
        console.warn('Geen gebruikers gevonden.');
        return;
      }

      const emailSelect = document.getElementById("emailSelect");
      
      // Controleer of Selectize al is geïnitialiseerd en haal de controle op
      let selectizeControl = $(emailSelect)[0].selectize;
      
      if (selectizeControl) {
        // Als Selectize al bestaat, maak dan de huidige selectie leeg
        selectizeControl.clear(); // Wis de geselecteerde items zonder de opties te wissen
      } else {
        // Initialiseer Selectize als het nog niet bestaat
        $(emailSelect).selectize({
          plugins: ['remove_button'],
          delimiter: ',',
          persist: false,
          create: false,
          sortField: 'text'
        });
        
        // Haal nu de selectize instance op
        selectizeControl = $(emailSelect)[0].selectize;
      }

      // Voeg opties toe aan de select dropdown
      selectizeControl.clearOptions(); // Zorg ervoor dat er geen dubbele opties komen te staan
      users.forEach(user => {
        if (user.data && user.data.Email && user.data.Email.trim() !== "") {
          selectizeControl.addOption({
            value: user.data.Email,
            text: user.data.Email
          });
        }
      });

      console.log('Gebruikers succesvol geladen en dropdown bijgewerkt met Selectize.');
      
      // Selecteer bestaande e-mails indien aanwezig
      if (selectedEmails.length > 0) {
        selectedEmails.forEach(email => {
          selectizeControl.addItem(email);
        });
      }
    })
    .catch(error => {
      console.error('Fout bij het laden van gebruikers:', error);
    });
}

**Comment by dvg-p4** [comment]
Due to a merge conflict between two incompatible feature requests, we now have the worst of both worlds: `clearOptions()` specifically *does not* remove currently-selected options; but *does* clear the selection. See #2146 for a more complete explanation.

**Comment by kunibertgangnam** [comment]
here is how i worked around this issue

edit: in my humble opinion there should **absolutely** be an option to keep all selected items but clear all options which are not selected

```
// do not use clearOptions here, as it also clears all selected items..
// also: currently there is a bug and clearOptions does not remove the options, but all selected items
//this.clearOptions();

// list of options to keep because they are selected
var optionsToKeep = [];
this.items.forEach((item) => {
	optionsToKeep.push(this.options[item]);
});

// list of options to remove
var optionsToRemove = [];
for (var [key, value] of Object.entries(this.options)) {
	if (!optionsToKeep.includes(value)){
		optionsToRemove.push(value);
	}
}

// remove options
optionsToRemove.forEach((option) => {
	this.removeOption(option.value, true);
});
```
